pkg/controllers/manageagent/manageagent.go

```diff
@@ -49,9 +49,9 @@ func Register(ctx context.Context,
 	relatedresource.WatchClusterScoped(ctx, "manage-agent-resolver", h.resolveNS, namespace, clusters)
 }
 
-func (h *handler) resolveNS(namespace, name string, obj runtime.Object) ([]relatedresource.Key, error) {
-	if _, ok := obj.(*fleet.Cluster); ok {
-		if _, err := h.bundleCache.Get(namespace, agentBundleName); err != nil {
+func (h *handler) resolveNS(namespace, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
+	if cluster, ok := obj.(*fleet.Cluster); ok {
+		if _, err := h.bundleCache.Get(namespace, name.SafeConcatName(agentBundleName, cluster.Name)); err == nil {
```

Comment:
I assumed that earlier if we didn't find bundle cache (err !=nil), we'd still return a key that would call agent-resolver and deploy agent. Now we assume agent bundle will always exist?

Reply:
Oh okay, so now importCluster will always deploy agent vs earlier it would do it only for manager initiated setups?

Reply:
whoops you raised a good question. I originall thought this was a bug. Like when we change the agentEnvs on the cluster, it didn't enqueue the namespace(which should cause resync) because we only check if `err != nil`. Now i think it is only when the bundle is not there then we enqueue the namespace.